src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/PrisonerService.kt

```diff
@@ -27,26 +27,36 @@ class PrisonerService(
     if (featureFlag.isNomisSearchEnabled()) {
       val prisoners = client.findPossibleMatches(PossibleMatchCriteria.from(person))
       log.debug("Number of prisoners returned from Nomis for PNC ${person.otherIdentifiers?.pncNumber} = ${prisoners?.size ?: 0}")
-      prisoners?.let {
-        if (matchesExistingPrisonerExactly(it, person)) {
+      prisoners?.let { prisonerList ->
+        if (pncIdentifierDoesNotMatch(prisonerList, person)) {
+          log.debug("Nomis PNC Id does not match that of person")
+          telemetryService.trackEvent(
+            TelemetryEventType.NOMIS_PNC_DOES_NOT_MATCH,
+            mapOf(
```

Comment:
I think we should also create a new person for non-matching PNC, no? 🤔

Reply:
Maybe, in order to match the behaviour of the HMCTS listener, although based on recent discussions it seems PNC may not be suitable as a unique identifier